tests/test_sharded_dataset.py

```diff
@@ -24,6 +25,51 @@ def test_sniff_format_for_json():
         assert _sniff_format_for_dataset(f.name) == ".json"
 
 
+def test_sniff_format_for_parquet():
+
+    import pyarrow as pa
+    import pyarrow.parquet as pq
+
+    with tempfile.NamedTemporaryFile(suffix=".parquet") as f:
+        table = pa.table({"col1": [1, 2, 3], "col2": ["a", "b", "c"]})
+        pq.write_table(table, f.name)
+        f.flush()
+
+        assert _sniff_format_for_dataset(f.name) == ".parquet"
+
+
 @skip_if_no_soundlibs
 def test_resolve_audio_pointer():
     AudioTextUrlDataSource.resolve_audio_pointer("https://ccrma.stanford.edu/~jos/mp3/trumpet.mp3", 16_000)
+
+
+def test_basic_parquet_datasource_read_row():
+
+    import pyarrow as pa
+    import pyarrow.parquet as pq
+
+    with tempfile.NamedTemporaryFile(suffix=".parquet", delete=False) as f:
+        # Create a simple dataset
+        data = {"column1": ["value1", "value2", "value3"], "column2": [10, 20, 30]}
+        table = pa.Table.from_pydict(data)
+        pq.write_table(table, f.name)
+
+    try:
```

Comment:
Why not put this within the context manager and set delete to True?

Reply:
Done!